pandas/io/excel/_odfreader.py

```diff
@@ -197,22 +197,24 @@ def _get_cell_string_value(self, cell) -> str:
         Find and decode OpenDocument text:s tags that represent
         a run length encoded sequence of space characters.
         """
-        from odf.element import Element, Text
+        from odf.element import Element
         from odf.namespaces import TEXTNS
-        from odf.text import P, S
+        from odf.text import S
 
-        text_p = P().qname
         text_s = S().qname
 
-        p = cell.childNodes[0]
-
         value = []
-        if p.qname == text_p:
-            for k, fragment in enumerate(p.childNodes):
-                if isinstance(fragment, Text):
-                    value.append(fragment.data)
-                elif isinstance(fragment, Element):
-                    if fragment.qname == text_s:
-                        spaces = int(fragment.attributes.get((TEXTNS, "c"), 1))
+
+        for fragment in cell.childNodes:
+            if isinstance(fragment, Element):
+                if fragment.qname == text_s:
+                    spaces = int(fragment.attributes.get((TEXTNS, "c"), 1))
                     value.append(" " * spaces)
+                else:
+                    # recursive impl needed in case of nested fragments
+                    # with multiple spaces
+                    # https://github.com/pandas-dev/pandas/pull/36175#discussion_r484639704
+                    value.append(self._get_cell_string_value(fragment))
+            else:
+                value.append(str(fragment))
```

Comment:
Is there any reason to call str(fragment) instead of always just calling `self._get_cell_string_value(fragment)`?

Reply:
then we'd need to handle the `str` method for all the different cases which would require going into the spec. `odfpy` already has the str implementations done. the only case that was problematic was the multiple spaces that was addressed in the previous linked PR.